File: Online-judge/leetcode/design-hashset.py
from collections import defaultdict


class MyHashSet:

    def __init__(self):
        self.dict = defaultdict(int)


    def add(self, key: int) -> None:
        self.dict[key] = True
        

    def remove(self, key: int) -> None:
        if key in self.dict.keys():
            del self.dict[key]
        

    def contains(self, key: int) -> bool:
        if key in self.dict.keys():
            return True
        else:
            return False


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)

File: Online-judge/leetcode/design-hashset.cpp
# include <bits/stdc++.h>
using namespace std;

/**
 * @author Md.Shohag <shohag.fks@gmail.com>
 * Problem : https://leetcode.com/problems/design-hashset/description/
 * Time complexity O(1)
 * Memory complexity O(1e6 + 1)
*/


class MyHashSet {
private:
    vector<int> bucket;
public:
    MyHashSet() {
        bucket.resize(1e6 + 1, 0);
    }
    
    void add(int key) {
        this->bucket[key] = 1;
    }
    
    void remove(int key) {
        this->bucket[key] = 0;
    }
    
    bool contains(int key) {
        return this->bucket[key];
    }
};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet* obj = new MyHashSet();
 * obj->add(key);
 * obj->remove(key);
 * bool param_3 = obj->contains(key);
 */